lib/helpers.py

```diff
@@ -54,6 +57,19 @@ def get_random_string(length=32, choices=None):
         choices = string.ascii_letters + string.digits
     return django_get_random_string(length=length, allowed_chars=choices)
 
+def get_hash(*params):
+    """
+    This function creates a hash of the given params.
+
+    @param length: The params that want to be hashed.
+    @return: Single hash that represent all the params given, a string
+    """
+    dhash = hashlib.md5()
+    for p in params:
+        param_encoded = json.dumps(p, sort_keys=True, default=str).encode()
```

Comment:
At first sight, I think the current backend code here and the frontend code chapter.js serialize the submission data differently to JSON. If the JSON string becomes different for the same submission data, then the output hash is also different. (They should be computed identically in the frontend and backend.)